skimage/transform/_geometric.py

```diff
@@ -39,24 +59,32 @@ def _center_and_normalize_points(points):
            (1997): 580-593.
 
     """
-
+    n, d = points.shape
     centroid = np.mean(points, axis=0)
 
-    rms = math.sqrt(np.sum((points - centroid) ** 2) / points.shape[0])
+    centered = points - centroid
+    rms = np.sqrt(np.sum(centered ** 2) / n)
+
+    # if all the points are the same, the transformation matrix cannot be
+    # created. We return an equivalent matrix with np.nans as sentinel values.
+    if rms == 0:
+        return np.full((d + 1, d + 1), np.nan), np.full_like(points, np.nan)
 
-    norm_factor = math.sqrt(2) / rms
+    norm_factor = np.sqrt(d) / rms
 
-    matrix = np.array([[norm_factor, 0, -norm_factor * centroid[0]],
-                       [0, norm_factor, -norm_factor * centroid[1]],
-                       [0, 0, 1]])
+    part_matrix = norm_factor * np.concatenate(
+            (np.eye(d), -centroid[:, np.newaxis]), axis=1
+            )
+    matrix = np.concatenate(
+            (part_matrix, [[0,] * d + [1]]), axis=0
+            )
```

Comment:
```suggestion
matrix = np.eye(d + 1)
matrix[:-1, -1] = -centroid[:, np.newaxis]
```

Reply:
@grlee77 this turns out to be incorrect — see `norm_factor *` above. Hooray for tests and git-bisect! :blush: Reverting this change.